src/documents/RemoteModelAPI.md

```diff
@@ -0,0 +1,316 @@
+---
+title: RemoteModel API
+layout: default
+category: Infusion
+---
+
+`fluid.remoteModelComponent` builds on top of [`fluid.modelComponent`](ComponentGrades.md) with the purpose of providing a buffer between a local and
+remote model that are attempting to stay in sync. For example a local model is being updated by user interaction, this is sent back
+to a remote server, which in turn tries to update the local model. If additional user actions occur during the roundtrip, an
+infinite loop of updates may occur. `fluid.remoteModelComponent` solves this by restricting reading and writing to a single request at a time,
+waiting for one request to complete before operating the next.
+
+## Supported Events
+
+In addition to the standard `fluid.modelCompnent` events, `fluid.remoteModelComponent` add the following:
+
+* `onFetch`,
+* `onFetchError`,
+* `afterFetch`,
+* `onWrite`,
+* `onWriteError`,
+* `afterWrite`
+
+For information on the different types of events, see [Infusion Event System](InfusionEventSystem.md).
+
+<div class="infusion-docs-note">
+    <strong>Note:</strong> The <code>onFetch</code>, <code>afterFetch</code>, <code>onWrite</code> and <code>afterWrite</code> events are synthetic events that fire as a sequence. As implemented in the <code>fluid.remoteModelComponent</code>, the listeners for the event will all fire in the specified order and be passed the same arguments. However, the next listener will not trigger till the previous one has completed. That is, either returned a value or resolved/rejected a returned promise. This is useful to prevent further actions from happening until all of the listeners have completed, but unlike other event sequences, it cannot be used to pass the payload through a series of transformations.
+</div>
+
+### onFetch
+
+<table>
+    <tbody>
+        <tr>
+            <th>Description</th>
+            <td>
+                Fires an event sequence the executes before <code>fetchImpl</code> is called.
+            </td>
+        </tr>
+        <tr>
+            <th>Type</th>
+            <td>default</td>
+        </tr>
+    </tbody>
+</table>
+
+### onFetchError
+
+<table>
+    <tbody>
+        <tr>
+            <th>Description</th>
+            <td>
+                Fired if the <code>onFetch</code> sequence or <code>fetchImpl</code> promises are rejected. By default the <code>afterFetch.unblock</code> (unblocks the  request queue) handler will execute.
+            </td>
+        </tr>
+        <tr>
+            <th>Type</th>
+            <td>default</td>
+        </tr>
+        <tr>
+            <th>Parameters</th>
+            <td>
+                Any paramaters passed along to the rejected promise. Typically this is an <code>error</code> object.
```

Comment:
Typo parameters. We need to give more guidance here on the expectations of the reject payload. Is it really typically an `error` object? Probably meant capital E `Error`. One convention we use a lot is for there to be an `isError: true` field if it is not an Error -

Reply:
@amb26 it is whatever gets returned for the rejected promise reason. So in that case, I don't really know exactly what it will be because it is dependent on what happens in the onFetch promise sequence and the onFetchImpl. Any thoughts on how to better document this?